builder/processor/src/main/java/io/helidon/builder/processor/GenerateAbstractBuilder.java

```diff
@@ -0,0 +1,1294 @@
+        generatePrototypeImpl(pw, typeContext, typeArgumentNames);
+
+        // end of BuilderBase class
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+    }
+
+    static void buildRuntimeObjectMethod(PrintWriter pw, TypeContext typeContext, boolean isBuilder) {
+        TypeContext.TypeInformation typeInformation = typeContext.typeInfo();
+        boolean hasRuntimeObject = typeInformation.runtimeObject().isPresent();
+        TypeName builtObject = typeInformation.runtimeObject().orElse(typeInformation.prototype());
+
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("@Override");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print("public ");
+        pw.print(builtObject.fqName());
+        pw.println(" build() {");
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.print("return ");
+        if (hasRuntimeObject) {
+            pw.print(builtObject.genericTypeName().fqName());
+            if (isBuilder) {
+                pw.println(".create(this.buildPrototype());");
+            } else {
+                pw.println(".create(this);");
+            }
+        } else {
+            if (isBuilder) {
+                pw.println("return build();");
+            } else {
+                pw.println("return this;");
+            }
+        }
+        pw.print(SOURCE_SPACING);
+        pw.print(SOURCE_SPACING);
+        pw.println("}");
+        pw.println();
+
+        // if impl, we also need to add the `get()` method from supplier
+        if (!isBuilder) {
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("@Override");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.print("public ");
+            pw.print(builtObject.fqName());
+            pw.println(" get() {");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("return build();");
+            pw.print(SOURCE_SPACING);
+            pw.print(SOURCE_SPACING);
+            pw.println("}");
+            pw.println();
+        }
+    }
+
+    static boolean hasConfig(List<PrototypeProperty> properties) {
+        return properties.stream()
+                .filter(it -> "config".equals(it.name()))
+                .anyMatch(it -> CONFIG_TYPE.equals(it.typeHandler().actualType()));
+    }
+
+    @SuppressWarnings("checkstyle:MethodLength") // will be fixed when we switch to model
+    private static void builderMethods(PrintWriter pw, TypeContext typeContext) {
+        List<PrototypeProperty> properties = typeContext.propertyData().properties();
+        TypeContext.ConfiguredData configured = typeContext.configuredData();
+
+        if (configured.configured() || hasConfig(properties)) {
+            /*
```

Comment:
these types of comments should be on the javadoc of the method instead of here.

Reply:
not really - this is a comment of a section of the code - the next section creates the `public Builder config(Config)` method, there is another section further down, that generates other builder methods. This is to help you to navigate to the exact code responsible for this, rather than looking at what is printed.